Bio/PDB/PDBList.py

```diff
@@ -479,17 +524,19 @@ def get_all_assemblies(self, file_format="mmCif"):
 
         # FTPLib is much faster than urlopen
         idx = self.pdb_server.find("://")
+        pdb_server = self.pdb_server
         if idx >= 0:
-            ftp = ftplib.FTP(self.pdb_server[idx + 3 :])
+            pdb_server = self.pdb_server[idx + 3 :]
+            ftp = ftplib.FTP(pdb_server.split("/", 1)[0])
         else:
-            ftp = ftplib.FTP(self.pdb_server)
+            ftp = ftplib.FTP(pdb_server.split("/", 1)[0])
         ftp.login()  # anonymous
-
+        pdb_directory = pdb_server.split("/", 1)[1]
         if file_format.lower() == "mmcif":
-            ftp.cwd("/pub/pdb/data/assemblies/mmCIF/all/")
+            ftp.cwd(f"{pdb_directory}/data/assemblies/mmCIF/all/")
```

Comment:
You can directly do `ftp.cwd(pdb_directory)` and then there's no need to attach it to the paths below.